src/java/com/zimbra/oauth/utilities/OAuth2ResourceUtilities.java

```diff
@@ -54,10 +54,10 @@
      * @acct the user account for which datasource is being setup
      * @throws ServiceException If there are issues
      */
-    public static final String authorize(String client, String relay, Account account) throws ServiceException {
+    public static final String authorize(String client, Map<String, String[]> params, Account account) throws ServiceException {
         final IOAuth2Handler oauth2Handler = ClassManager.getHandler(client);
-        ZimbraLog.extensions.debug("Client : %s, handler:%s, relay:%s ", client, oauth2Handler,relay);
-        return oauth2Handler.authorize(relay, account);
+        ZimbraLog.extensions.debug("Client : %s, handler:%s, relay:%s, type:%s ", client, oauth2Handler, params.get("relay"), params.get("type"));
+        return oauth2Handler.authorize(params, account);
```

Comment:
Do we want to filter the params with the method:
https://github.com/Zimbra/zm-oauth-social/blob/develop/src/java/com/zimbra/oauth/utilities/OAuth2ResourceUtilities.java#L140

Same as is done in `OAuth2ResourceUtilities#authenticate`.

This would also remove the need to deal with an array for single value elements in `authorize`.